core/src/main/scala/org/apache/spark/util/SignalUtils.scala

```diff
@@ -31,14 +31,51 @@ import org.apache.spark.internal.Logging
 /**
  * Contains utilities for working with posix signals.
  */
-private[spark] object Signaling extends Logging {
+private[spark] object SignalUtils extends Logging {
+
+  private var registered = false
```

Comment:
since SignalLogging  is being merged with SignalUtils, it might be better to give this a more specific name such  `loggerRegistered`